app/src/main/java/it/quip/android/model/BaseParseObject.java

```diff
@@ -0,0 +1,25 @@
+package it.quip.android.model;
+
+import com.parse.ParseObject;
+
+import it.quip.android.util.MockUtils;
+
+
+public class BaseParseObject extends ParseObject {
+
+    protected void safePut(String key, Object value) {
+        if (value != null) {
+            this.put(key, value);
+        }
+    }
+
+    @Override
+    public String getObjectId() {
+        String objectId = super.getObjectId();
+        if (null == objectId) {
+            objectId = MockUtils.randomId();
```

Comment:
Do we really want to do this? Having no object id should signify that it hasn't been saved yet